core/src/test/scala/org/apache/spark/sql/delta/DeltaSuite.scala

```diff
@@ -873,113 +904,106 @@ class DeltaSuite extends QueryTest
           .mode("append")
           .save(tempDir.getCanonicalPath)
 
-        val e = intercept[AnalysisException] {
+        val e = intercept[IllegalArgumentException] {
           Seq((3, "x"), (5, "x")).toDF("value", "part2")
             .withColumn("part1", $"value" % 2)
             .write
             .format("delta")
             .partitionBy("part1", "part2")
             .mode("overwrite")
-            .option(DeltaOptions.REPLACE_WHERE_OPTION, "part1 = 1")
             .option(DeltaOptions.PARTITION_OVERWRITE_MODE_OPTION, "dynamic")
+            .option(DeltaOptions.REPLACE_WHERE_OPTION, "part1 = 1")
             .save(tempDir.getCanonicalPath)
         }
-        assert(e.getMessage ===
-          "'replaceWhere' cannot be used when 'partitionOverwriteMode' is set to 'DYNAMIC'")
+        assert(e.getMessage === "A 'replaceWhere' expression and " +
+          "'partitionOverwriteMode'='dynamic' cannot both be set in the DataFrameWriter options.")
       }
     }
   }
 
-  test(
-    "batch write: append, dynamic partition overwrite conf not supported with replaceWhere") {
-    withSQLConf(DeltaSQLConf.DYNAMIC_PARTITION_OVERWRITE_ENABLED.key -> "true") {
-      withSQLConf(SQLConf.PARTITION_OVERWRITE_MODE.key -> "dynamic") {
-        withTempDir { tempDir =>
-          def data: DataFrame = spark.read.format("delta").load(tempDir.toString)
```

Comment:
this test is removed and replaced by the above two tests.

each following test name, and contents diffs are shifted by one.
I just fixed some style issues and since that messed with the indents the diff is messed up.